buildSrc/src/main/kotlin/KotlinConfiguration.kt

```diff
@@ -83,3 +82,48 @@ fun getOverridingKotlinApiVersion(project: Project): String? {
  */
 fun irValidationMode(project: Project): String? =
     project.providers.gradleProperty("kotlin_ir_validation_mode").orNull
+
+/**
+ * Unconditional compiler flags required of Kotlin User Projects. See KT-75078.
+ */
+fun KotlinCommonCompilerOptions.addKotlinUserProjectFlags() {
+    freeCompilerArgs.addAll(
+        "-Xreport-all-warnings",
+        "-Xrender-internal-diagnostic-names",
+    )
+}
+
+/**
+ * Use `kotlin_Werror_override` property to override the state of -Werror:
+ * `true` means that warnings should be treated as errors,`false` means that they should not.
+ */
+private fun warningsAreErrorsOverride(project: Project): Boolean? =
+    when (val prop = project.rootProject.properties["kotlin_Werror_override"] as? String) {
```

Comment:
It's better to use `providers.gradleProperty` instead of `project.rootProject.properties`, see [the changes](https://github.com/Kotlin/kotlinx.serialization/pull/2997/files#diff-09e3a69c8143d7af22e62f518f556102bbdea00485f9e30be972303bc6c7dc19R8) in the serialization

Reply:
Fixed that